File: sdk/typescript/src/knowledge.ts
/**
 * TypeScript SDK client for Phase 5.46 Knowledge Assurance platform.
 */

export interface KnowledgeReference {
  reference_id: string;
  tenant_id: string;
  external_key: string;
  resource_type: string;
  status: string;
  classification: string;
  sha256_checksum: string;
  is_immutable: boolean;
  created_at: string;
}

export interface KnowledgeTrustAssessment {
  assessment_id: string;
  tenant_id: string;
  target_resource_id: string;
  overall_score: number;
  trust_band: string;
  confidence_level: string;
  created_at: string;
}

export class KnowledgeAssuranceClient {
  constructor(private baseUrl: string, private apiKey?: string) {}

  private async request(path: string, options: RequestInit = {}, tenantId = "default_tenant"): Promise<any> {
    const headers: Record<string, string> = {
      "Content-Type": "application/json",
      "x-tenant-id": tenantId,
      ...(this.apiKey ? { Authorization: `Bearer ${this.apiKey}` } : {}),
      ...(options.headers as Record<string, string>),
    };

    const res = await fetch(`${this.baseUrl}${path}`, { ...options, headers });
    if (!res.ok) {
      throw new Error(`Knowledge Assurance API error: ${res.statusText}`);
    }
    return res.json();
  }

  async getStatus(tenantId = "default_tenant"): Promise<any> {
    return this.request("/v1/knowledge/status", { method: "GET" }, tenantId);
  }

  async createReference(
    externalKey: string,
    resourceType = "DOCUMENT",
    classification = "INTERNAL",
    tenantId = "default_tenant"
  ): Promise<KnowledgeReference> {
    return this.request(
      "/v1/knowledge/references",
      {
        method: "POST",
        body: JSON.stringify({ external_key: externalKey, resource_type: resourceType, classification }),
      },
      tenantId
    );
  }

  async listReferences(tenantId = "default_tenant"): Promise<KnowledgeReference[]> {
    return this.request("/v1/knowledge/references", { method: "GET" }, tenantId);
  }

  async registerSource(
    name: string,
    sourceType = "DOCUMENT_REPOSITORY",
    authority = "AUTHORITATIVE",
    tenantId = "default_tenant"
  ): Promise<any> {
    return this.request(
      "/v1/knowledge/sources",
      {
        method: "POST",
        body: JSON.stringify({ name, source_type: sourceType, authority }),
      },
      tenantId
    );
  }

  async evaluateTrust(targetResourceId: string, tenantId = "default_tenant"): Promise<KnowledgeTrustAssessment> {
    return this.request(`/v1/knowledge/trust/${targetResourceId}`, { method: "GET" }, tenantId);
  }

  async assessAssurance(targetResourceId: string, tenantId = "default_tenant"): Promise<any> {
    return this.request(`/v1/knowledge/assurance/${targetResourceId}`, { method: "GET" }, tenantId);
  }

  async getAnalyticsReport(tenantId = "default_tenant"): Promise<any> {
    return this.request("/v1/knowledge/analytics/report", { method: "GET" }, tenantId);
  }
}

export class KnowledgeClient {
  constructor(
    private fetchFn: typeof fetch,
    private baseUrl: string,
    private headers: Record<string, string>
  ) {}

  async search(query: string): Promise<any> {
    const res = await this.fetchFn(`${this.baseUrl}/v1/knowledge/search`, {
      method: 'POST',
      headers: this.headers,
      body: JSON.stringify({ query }),
    });
    return res.json();
  }
}

File: sdk/go/knowledge.go
package llmengine

import (
	"bytes"
	"encoding/json"
	"fmt"
	"net/http"
)

type KnowledgeReference struct {
	ReferenceID    string `json:"reference_id"`
	TenantID       string `json:"tenant_id"`
	ExternalKey    string `json:"external_key"`
	ResourceType   string `json:"resource_type"`
	Status         string `json:"status"`
	Classification string `json:"classification"`
	SHA256Checksum string `json:"sha256_checksum"`
	IsImmutable    bool   `json:"is_immutable"`
}

type KnowledgeTrustAssessment struct {
	AssessmentID     string  `json:"assessment_id"`
	TenantID         string  `json:"tenant_id"`
	TargetResourceID string  `json:"target_resource_id"`
	OverallScore     float64 `json:"overall_score"`
	TrustBand        string  `json:"trust_band"`
}

type KnowledgeAssuranceService struct {
	client *Client
}

func (s *KnowledgeAssuranceService) GetStatus(tenantID string) (map[string]interface{}, error) {
	req, err := http.NewRequest("GET", fmt.Sprintf("%s/v1/knowledge/status", s.client.BaseURL), nil)
	if err != nil {
		return nil, err
	}
	req.Header.Set("x-tenant-id", tenantID)

	resp, err := s.client.HTTPClient.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	if resp.StatusCode >= 400 {
		return nil, fmt.Errorf("knowledge status API error: %s", resp.Status)
	}

	var result map[string]interface{}
	if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
		return nil, err
	}
	return result, nil
}

func (s *KnowledgeAssuranceService) CreateReference(tenantID, externalKey, resourceType, classification string) (*KnowledgeReference, error) {
	reqBody := map[string]interface{}{
		"external_key":   externalKey,
		"resource_type":  resourceType,
		"classification": classification,
	}
	bodyBytes, _ := json.Marshal(reqBody)

	req, err := http.NewRequest("POST", fmt.Sprintf("%s/v1/knowledge/references", s.client.BaseURL), bytes.NewBuffer(bodyBytes))
	if err != nil {
		return nil, err
	}
	req.Header.Set("Content-Type", "application/json")
	req.Header.Set("x-tenant-id", tenantID)

	resp, err := s.client.HTTPClient.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	if resp.StatusCode >= 400 {
		return nil, fmt.Errorf("knowledge reference API error: %s", resp.Status)
	}

	var ref KnowledgeReference
	if err := json.NewDecoder(resp.Body).Decode(&ref); err != nil {
		return nil, err
	}
	return &ref, nil
}

func (s *KnowledgeAssuranceService) EvaluateTrust(tenantID, targetResourceID string) (*KnowledgeTrustAssessment, error) {
	req, err := http.NewRequest("GET", fmt.Sprintf("%s/v1/knowledge/trust/%s", s.client.BaseURL, targetResourceID), nil)
	if err != nil {
		return nil, err
	}
	req.Header.Set("x-tenant-id", tenantID)

	resp, err := s.client.HTTPClient.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	if resp.StatusCode >= 400 {
		return nil, fmt.Errorf("knowledge trust API error: %s", resp.Status)
	}

	var trust KnowledgeTrustAssessment
	if err := json.NewDecoder(resp.Body).Decode(&trust); err != nil {
		return nil, err
	}
	return &trust, nil
}
